### PR TITLE
Bsr use pylint in tests

### PR DESCRIPTION
**Pull request by cnormant-pass**
## But de la pull request

Réactiver pylint pour les tests

## Vérifications

- [ ] J'ai écrit les tests nécessaires
- [ ] J'ai mis à jour le fichier des [plans de tests](https://docs.google.com/spreadsheets/d/12I9f68L312xEE8lKFN7LsBHO2M_tcBBMSs0Be6qCQ98/edit) du portail pro si nécessaire
- [ ] J'ai mis à jour [la liste des routes et des titres](https://www.notion.so/passcultureapp/Titre-des-pages-de-l-espace-Pro-f4e490619bc54010adeb67c86d5e6a40?pvs=4) de pages du portail pro si j'en ai rajouté/modifié ou supprimé une.
- [ ] J'ai [relu attentivement les migrations](https://www.notion.so/passcultureapp/Clarifier-les-pratiques-de-migration-de-BDD-5f8edeba57ed4a17b80c847a74def027), en particulier pour éviter les _locks_, et je préviens les équipes Shérif et Data
- [ ] J'ai ajouté des screenshots pour d'éventuels changements graphiques
- [ ] J'ai fait la revue fonctionnelle de mon ticket

### CODE REVIEW

File: api/tests/core/offers/test_repository.py [hunk 1]
```diff
@@ -2642,24 +2642,23 @@ def test_get_inactive_headline_offers_basic(self):
 
         active_offer = factories.OfferFactory(isActive=True)
         factories.StockFactory(offer=active_offer)
-        active_offer_headline_offer = factories.HeadlineOfferFactory(offer=active_offer)
+        # active offer headline offer:
+        factories.HeadlineOfferFactory(offer=active_offer)
```
Comment (jcicurel-pass) [inline api/tests/core/offers/test_repository.py:2646]: Par curiosité, est ce que tu as toujours une erreur si tu écris par exemple `_active_offer_headline_offer = ...` ? Si ça empêche l'erreur ça peut remplacer le commentaire éventuellement

File: api/tests/core/external/pro_user_automations_test.py [hunk 2]
```diff
@@ -63,7 +63,7 @@ def test_automation(self, with_subaccount):
 
             with patch(self.MOCK_IMPORT_CONTACT_PATH) as mock_import_contacts:
                 with override_features(WIP_ENABLE_BREVO_PRO_SUBACCOUNT=with_subaccount):
-                    assert type(self).func()
+                    assert type(self).func()  # pylint: disable=no-value-for-parameter # I don't get this code
```
Comment (jcicurel-pass) [inline api/tests/core/external/pro_user_automations_test.py:66]: ah ouais il est mystique celui là
pas d'indice dans des sous-classes ?

File: api/tests/core/external_bookings/test_decorator.py [hunk 3]
```diff
@@ -126,7 +131,9 @@ def test_should_raise_understandable_error_if_the_decorated_func_is_called_with_
             ),
         ],
     )
-    def test_should_log_error(self, args_list, kwargs_dict, exception_class, request_mock, expected_extra, caplog):
+    def test_should_log_error(  # pylint: disable=too-many-positional-arguments
```
Comment (jcicurel-pass) [inline api/tests/core/external_bookings/test_decorator.py:134]: y avait pas moyen d'ignorer cette erreur pour tous les tests c'est ça ?

File: api/tests/routes/public/collective/endpoints/adage_mock/test_bookings.py [hunk 4]
```diff
@@ -696,7 +696,10 @@ def test_cannot_book_offer_if_no_educational_year_matches_beginning(self, client
             # deposit creates an educational year
             collectiveOffer__institution=deposit.educationalInstitution,
             # beginning will not match the educational year
-            beginningDatetime=datetime.now(timezone.utc) + timedelta(weeks=512),
+            beginningDatetime=datetime.now(  # pylint: disable=datetime-now # ok because we use timezone aware datetime
```
Comment (jcicurel-pass) [inline api/tests/routes/public/collective/endpoints/adage_mock/test_bookings.py:699]: Elle a vraiment un intérêt cette règle ? Pour moi si on écrit ça c'est qu'on veut un now aware, ça me parait assez clair

File: api/pyproject.toml [hunk 5]
```diff
@@ -173,49 +173,45 @@ disable_error_code = ["truthy-function"]
...
+  "singleton-comparison",
+  "superfluous-parens",
+  "too-few-public-methods",
+  "too-many-ancestors",
+  "too-many-arguments",
+  "too-many-branches",
+  "too-many-instance-attributes",
+  "too-many-lines",
+  "too-many-locals",
+    "too-many-positional-arguments",  # mainly for the test folder
```
Comment (jcicurel-pass) [inline api/pyproject.toml:205]: Il y a un indent en trop je crois ?

ça sert à quoi cette section d'ailleurs ?

**Review by jcicurel-pass — COMMENTED** [review]
Bien jouej 👌

File: api/tests/core/external/pro_user_automations_test.py (hunk 2 above)
Comment (vroullier-pass) [reply api/tests/core/external/pro_user_automations_test.py:66]: Les 4 valeurs possibles de func ne prennent pas de paramètre...

**Review by vroullier-pass — APPROVED** [review]
Nice

File: api/tests/core/offers/test_repository.py (hunk 1 above)
Comment (cnormant-pass) [reply api/tests/core/offers/test_repository.py:2646]: Oui pylint warn même avec des variables `_variable`, j'avais essayé ^^

File: api/tests/core/external/pro_user_automations_test.py (hunk 2 above)
Comment (cnormant-pass) [reply api/tests/core/external/pro_user_automations_test.py:66]: Stack overflow voulait que j'ajoute un `__init__` pour attribuer des valeurs à `self` mais ça ne changeait pas le warning
C'est suffisamment edge case pour ignorer, je pense, mais preneur d'idées !

File: api/tests/core/external_bookings/test_decorator.py (hunk 3 above)
Comment (cnormant-pass) [reply api/tests/core/external_bookings/test_decorator.py:134]: Ah si je l'ai ignorée (pour tout le code, pas moyen pour uniquement un sous dossier)
Je vais retirer ces ignore là

File: api/tests/routes/public/collective/endpoints/adage_mock/test_bookings.py (hunk 4 above)
Comment (cnormant-pass) [reply api/tests/routes/public/collective/endpoints/adage_mock/test_bookings.py:699]: Elle est custom, elle avait un intérêt, avant, maintenant que .utcnow() est déprécié elle va disparaitre au profit de cette syntaxe, donc j'ai ignore sans creuser, vu que ça va sauter un jour prochain

File: api/pyproject.toml (hunk 5 above)
Comment (cnormant-pass) [reply api/pyproject.toml:205]: Les warning qu'on ignore sur tout le code
yes je sais pas d'où sort cet indent ^^

File: api/tests/core/external_bookings/test_decorator.py (hunk 3 above)
Comment (jcicurel-pass) [reply api/tests/core/external_bookings/test_decorator.py:134]: Ah mais on voulait pas l'ignorer pour tout le code si ?

File: api/tests/routes/public/collective/endpoints/adage_mock/test_bookings.py (hunk 4 above)
Comment (jcicurel-pass) [reply api/tests/routes/public/collective/endpoints/adage_mock/test_bookings.py:699]: ça roule

File: api/tests/core/external_bookings/test_decorator.py (hunk 3 above)
Comment (cnormant-pass) [reply api/tests/core/external_bookings/test_decorator.py:134]: Non 😅 
C'est quand même un meilleur compromis, la règle à raison, mais est assez simple à voir 
Alors que pas pylint dans les tests a plus de conséquences (tests non lancés notamment)

**Review by jcicurel-pass — APPROVED** [review]

**Comment by cnormant-pass** [comment]
Découpé en PR liées
